src/app/ReadHandler.h

```diff
@@ -117,17 +124,25 @@ class ReadHandler : public Messaging::ExchangeDelegate
     // is larger than current self vended event number
     void MoveToNextScheduledDirtyPriority();
 
+    bool IsSubscription() { return mInteractionType == InteractionType::Subscribe; }
     bool IsInitialReport() { return mInitialReport; }
+    CHIP_ERROR OnSubscribeRequest(Messaging::ExchangeContext * apExchangeContext, System::PacketBufferHandle && aPayload);
+    void GetSubscriptionId(uint64_t & aSubscriptionId) { aSubscriptionId = mSubscriptionId; }
 
 private:
     enum class HandlerState
     {
-        Uninitialized = 0, ///< The handler has not been initialized
-        Initialized,       ///< The handler has been initialized and is ready
-        Reportable,        ///< The handler has received read request and is waiting for the data to send to be available
-        Reporting,         ///< The handler is reporting
+        Uninitialized = 0,     ///< The handler has not been initialized
+        Initialized,           ///< The handler has been initialized and is ready
+        Reportable,            ///< The handler has received read request and is waiting for the data to send to be available
```

Comment:
Reportable as a term doesn't describe a state, but a capability.  Would recommend the following:

```suggestion
        GeneratingReports,            ///< The handler has received either a Read or Subscribe request and is the process of generating a report.
```

Reply:
Done, thanks